### PR TITLE
Test foundry

### PR DESCRIPTION
**Pull request by volkanguneri**
tested all OnChainAIv1 setter functions' ownership access & events
Capture

### CODE REVIEW

File: foundry/test/OnChainAIv1.t.sol [hunk 1]
```diff
@@ -1,26 +1,105 @@
 // SPDX-License-Identifier: UNLICENSED
-pragma solidity ^0.8.13;
+pragma solidity 0.8.28;
 
 import {Test, console} from "forge-std/Test.sol";
+import {DeployLite} from "@forge-deploy-lite/DeployLite.s.sol";
```
Comment (zapaz) [inline foundry/test/OnChainAIv1.t.sol]: no need to re-write the script to deploy OnChainAI, just use it as before

File: foundry/test/OnChainAIv1.t.sol [hunk 2]
```diff
@@ -1,26 +1,105 @@
...
+        string memory javascript = vm.readFile("../chainlink/openai/OnChainAI.js");
+        uint32 gasLimit = 300000;
+        setJsonFile("../chainlink/config.json");
+        address router = readAddress("router");
+        uint64 subscriptionId = uint64(readUint("subscriptionId"));
+        string memory donId = readString("donId");
+        bytes32 donIdHex = bytes32(abi.encodePacked(donId));
+        uint256 price = readUint("price");
+        setJsonFile("");
+
```
Comment (zapaz) [inline foundry/test/OnChainAIv1.t.sol]: no reason to modify

File: foundry/test/OnChainAIv1.t.sol [hunk 3]
```diff
@@ -1,26 +1,105 @@
...
+        vm.expectRevert();
+        onChainAIv1.withdraw();
+
+        vm.prank(owner);
+        onChainAIv1.withdraw();
+
+        assert(address(onChainAIv1).balance == 0);
+        assert(address(owner).balance == initialAmount);
     }
```
Comment (zapaz) [inline foundry/test/OnChainAIv1.t.sol]: test_withdraw perfect :-)

File: foundry/test/OnChainAIv1.t.sol [hunk 4]
```diff
@@ -1,26 +1,105 @@
...
+   
+        vm.expectEmit(true, true, true, true);
+        emit JavascriptLog(newJavascript_);
+
+        vm.expectRevert();
+        onChainAIv1.setJavascript(newJavascript_);
+
+        vm.prank(owner);
+        onChainAIv1.setJavascript(newJavascript_);
+    }
```
Comment (zapaz) [inline foundry/test/OnChainAIv1.t.sol:97]: missing assert statement

assert that `javascript` stored is `javascript` given as parameter

something like : 
`assert(onChainAIv1.javascript == newJavascript_);`

is storage not accessible, `harness` the smartcontract, use a Test Class of OnChainAI :
`OnChainAIv1Test is OnChainAI`
and add public (or external) function `readJavascript`

this make require (?) not take into account first remark on rewrite deploy

File: foundry/test/OnChainAIv1.t.sol [hunk 5]
```diff
@@ -1,26 +1,105 @@
...
+
+    function test_setSubscriptionId(uint64 subscriptionId_) public {
+        vm.expectRevert();
+        onChainAIv1.setSubscriptionId(subscriptionId_);
+
+        vm.prank(owner);
+        onChainAIv1.setSubscriptionId(subscriptionId_);
+    }
```
Comment (zapaz) [inline foundry/test/OnChainAIv1.t.sol:107]: idem , and in all following set functions

**Review by zapaz — COMMENTED** [review]
main component of a test file is the assertion `assert(...)`  
in general must have at least one per test 

when internal storage not accessible, must harness smartcontract
nethertheless not very easy with private storage (better avoid it in original code)

**Comment by volkanguneri** [comment]
added harness functions and assertions to test all setter functions of OnChainAIV1.

**Review by zapaz — APPROVED** [review]

**Pull request merged**